Make Launcher::launchWithExecutable() protected

mikenz opened the pull request:
This allows us to extend Launcher. I need a custom launch() command to add some extra args and to specify the executable location.

mikenz committed:
Make Launcher::launchWithExecutable() protected

This allows us to extend Launcher. I need a custom launch() command to add some extra args and to specify the executable location.

jakubkulhan commented:
Hi, thank you for your PR! I see your use case. However, I don't think that making method `protected` and extending class is the right way. I think it would be better to make method `public`. Or even better to have `$executable` property like `$workDir`/`$env`/... that `launch` would use. I will be happy to merge a change like that.

mikenz committed:
Merge branch 'master' of https://github.com/jakubkulhan/chrome-devtools-protocol into patch-1

mikenz committed:
debug output

mikenz committed:
Remove debugging

mikenz committed:
Add debuging to find
 TypeError: Argument 2 passed to ChromeDevtoolsProtocol\Context::withDeadline() must be an instance of DateTimeImmutable, boolean given, called in vendor/jakubkulhan/chrome-devtools-protocol/src/ChromeDevtoolsProtocol/Context.php on line 41

mikenz committed:
Ensure the deadlineTimestamp always has a decimal component

mikenz committed:
Make sure we recieved something before attempting to iterate over it

jakubkulhan commented:
solved by #10
